src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -281,6 +284,17 @@
         return Uni.createFrom().voidItem();
     }
 
+    private <T extends EndpointProperties> Uni<List<T>> find(Class<T> typedEndpointClass, Set<UUID> endpointIds, boolean useStatelessSession) {
+        if (useStatelessSession) {
+            String query = "FROM " + typedEndpointClass.getSimpleName() + " WHERE id IN (:endpointIds)";
```

Comment:
I didn't notice before that we have to use HQL always in the stateless sessions.

Reply:
Not always, there are a [few methods](https://hibernate.org/reactive/documentation/1.0/reference/html_single/#_stateless_sessions) to persist/retrieve entities without HQL. But unfortunately, there is no stateless version of `session.find(typedEndpointClass, endpointIds.toArray())` so this one has to be replaced with HQL.